### PR TITLE
RHDEVDOCS-2802 Jenkins shared library in Tekton + RHDEVDOCS-2875 Basic common use cases

### PR DESCRIPTION
**Pull request by sounix000**
- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: `enterprise-4.9`, `enterprise-4.10`, `enterprise-4.11`
- **JIRA issues**: 
  - [RHDEVDOCS-2802 Dealing with Jenkins shared library in Tekton](https://issues.redhat.com/browse/RHDEVDOCS-2802)
  - [RHDEVDOCS-2875 Document basic common use cases](https://issues.redhat.com/browse/RHDEVDOCS-2875)
- **Preview pages**: [Examples of common use cases](https://deploy-preview-42629--osdocs.netlify.app/openshift-enterprise/latest/cicd/jenkins-tekton/migrating-from-jenkins-to-tekton#jt-examples-of-common-use-cases_migrating-from-jenkins-to-tekton)
- **SME review**: @rupalibehera      
- **Peer-review**: @jc-berger

### CODE REVIEW

File: modules/jt-examples-of-common-use-cases.adoc [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            oc process -f jboss-eap70-binary-build.json | oc create -f -
+            oc start-build eap-helloworld-app --from-dir=artifacts/
+            oc new-app jboss-eap70-deploy.json
+
+----
+
+== Extending the core capabilities of Jenkins and Tekton by using plugins
+Jenkins has the advantage of a very rich ecosystem of numerous plugins developed over the years by its extensive user base. You can search and browse the plugins in the link:https://plugins.jenkins.io/[Jenkins Plugin Index]. 
+
+The comparatively newer Tekton also has many tasks developed and contributed by the community and enterprise users. A publicly available catalogue of reusable Tekton tasks are available in the link:https://hub.tekton.dev/[Tekton Hub].
```
Comment (rupalibehera) [inline modules/jt-examples-of-common-use-cases.adoc:159]:
```suggestion
The comparatively newer Tekton also has many tasks developed and contributed by the community and enterprise users. A publicly available catalog of reusable Tekton tasks are available in the link:https://hub.tekton.dev/[Tekton Hub].
```
Should it be catalog instead because we say it Tekton catalog : https://github.com/tektoncd/catalog

File: modules/jt-examples-of-common-use-cases.adoc [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+In addition, Tekton incorporates many of the plugins of the Jenkins ecosystem within its core capabilities. For example, authorization is a critical function in both Jenkins and Tekton. While Jenkins ensures authorization using the link:https://plugins.jenkins.io/role-strategy/[Role-based Authorization Strategy] plugin, Tekton has built-in xref:../../authentication/using-rbac.adoc#using-rbac[Role-based Access Control] (RBAC).
+
+== Sharing reusable code in Jenkins and Tekton
+Jenkins link:https://www.jenkins.io/doc/book/pipeline/shared-libraries/[shared libraries] provide reusable code for parts of Jenkins pipelines. The libraries are shared between link:https://www.jenkins.io/doc/book/pipeline/jenkinsfile/[Jenkinsfiles] to create highly modular pipelines without code repetition.
+
+Although there is no direct equivalent of Jenkins shared libraries in Tekton, you can achieve similar workflows by using tasks from the link:https://hub.tekton.dev/[Tekton Hub], in combination with xref:./jt-extending-tekton-capabilities-using-custom-tasks-and-scripts.adoc[custom tasks and scripts].
```
Comment (rupalibehera) [inline modules/jt-examples-of-common-use-cases.adoc:166]: ./jt-extending-tekton-capabilities-using-custom-tasks-and-scripts.adoc this link seems to be 404 here https://deploy-preview-42629--osdocs.netlify.app/openshift-enterprise/latest/cicd/jenkins-tekton/migrating-from-jenkins-to-tekton#jt-examples-of-common-use-cases_migrating-from-jenkins-to-tekton

Comment (rupalibehera) [reply modules/jt-examples-of-common-use-cases.adoc:166]: oh is it because it is referring to our existing docs

File: modules/jt-examples-of-common-use-cases.adoc [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            oc new-app jboss-eap70-deploy.json
+
+----
+
+== Extending the core capabilities of Jenkins and Tekton by using plugins
+Jenkins has the advantage of a very rich ecosystem of numerous plugins developed over the years by its extensive user base. You can search and browse the plugins in the link:https://plugins.jenkins.io/[Jenkins Plugin Index]. 
+
+The comparatively newer Tekton also has many tasks developed and contributed by the community and enterprise users. A publicly available catalogue of reusable Tekton tasks are available in the link:https://hub.tekton.dev/[Tekton Hub].
+
+In addition, Tekton incorporates many of the plugins of the Jenkins ecosystem within its core capabilities. For example, authorization is a critical function in both Jenkins and Tekton. While Jenkins ensures authorization using the link:https://plugins.jenkins.io/role-strategy/[Role-based Authorization Strategy] plugin, Tekton has built-in xref:../../authentication/using-rbac.adoc#using-rbac[Role-based Access Control] (RBAC).
```
Comment (rupalibehera) [inline modules/jt-examples-of-common-use-cases.adoc:161]:
```suggestion
In addition, Tekton incorporates many of the plugins of the Jenkins ecosystem within its core capabilities. For example, authorization is a critical function in both Jenkins and Tekton. While Jenkins ensures authorization using the link:https://plugins.jenkins.io/role-strategy/[Role-based Authorization Strategy] plugin, Tekton leverages on Openshift's built-in xref:../../authentication/using-rbac.adoc#using-rbac[Role-based Access Control] (RBAC) system.
```

File: modules/jt-examples-of-common-use-cases.adoc (hunk 1 above)
Comment (sounix000) [reply modules/jt-examples-of-common-use-cases.adoc:159]: Yeah, my bad :-( . In spite of writing American English professionally for so many years now, I still make these mistakes.

**Review by rupalibehera — APPROVED** [review]

File: modules/jt-examples-of-common-use-cases.adoc [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+[id="jt-examples-of-common-use-cases_{context}"]
+= Examples of common use cases
+
+Both Jenkins and Tekton offer capabilities for common CI/CD use cases, such as: 
+
+* Compiling, building images, and deploying images
+* Extending the core capabilities by using plugins
+* Reusing shareable libraries and custom scripts
+
+This section briefly describes the comparable workflows for some common use cases in Jenkins and Tekton.
```
Comment (jc-berger) [inline modules/jt-examples-of-common-use-cases.adoc:15]: IBM SG page 22: avoid self referential language, like `This section describes...` 

I think this entire sentence can be omitted. I think seeing the examples of use cases sets the users up to continue reading documentation to learn more about the use cases and capabilities.

File: modules/jt-examples-of-common-use-cases.adoc [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+== Running a maven pipeline in Jenkins and Tekton
+
+Maven is used extensively in the Jenkins workflows. You can map your existing workflow to Tekton by using the following examples:
```
Comment (jc-berger) [inline modules/jt-examples-of-common-use-cases.adoc:19]: Not that I think passive voice is a big deal here, but if you did want to avoid it, you could rephrase it to something like:

- `Maven has an extensive role int eh Jenkins workflow.` 
- `The Jenkins workflow extensively incorporates Maven.` 
- `The Jenkins workflow includes/incorporates/utilizes/uses Maven.`

Comment (jc-berger) [reply modules/jt-examples-of-common-use-cases.adoc:19]: I'm a little lost about the connection between these two sentences. The title does state `Running a maven pipeline in Jenkins and Tekton` so I expect to learn about all these topics. 

However, the first sentence tells me maven is used extensively in the Jenkins workflow. Then the next sentence tells me to map my workflow to Tekton, and use the examples to map my workflow. 

So, am I focusing my efforts to map my workflow to Tekton or learn about maven in Jenkins? Or both, in which case, how does maven in Jenkins workflow connect to Tekton workflows? 

Would this connection work?:

`Maven has an extensive role  in the Jenkins workflow. You can map your existing Jenkins workflow to Tekton by using the following examples:`

I'm not sure if the above suggestion is correct because I'm not sure if the workflows the users are mapping are Jenkins workflows. Probably not because the first sentence informs users that maven is extensive in Jenkins workflows. So if users don't know that, they're not familiar with Jenkins workflows....Basically, it seems like we jumped from Jenkins to Tekton without a segway.

File: modules/jt-examples-of-common-use-cases.adoc [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+.Example: Compile, build an image, and deploy to OpenShift using maven on Jenkins
```
Comment (jc-berger) [inline modules/jt-examples-of-common-use-cases.adoc:21]: How about, `Example: Compile and build an image and deploy it to OpenShift using maven on Jenkins` 

The original sentence construction makes me think I'm doing three separate actions: 1) compiling (not sure what), 2) building an image, 3) deploying the image to OpenShift.

File: modules/jt-examples-of-common-use-cases.adoc [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    sh 'oc project helloworldproject'
+    sh 'oc process -f helloworld/jboss-eap70-binary-build.json | oc create -f -'
+    sh 'oc start-build eap-helloworld-app --from-dir=artifacts/'
+ 
+    stage 'Deploy'
+    sh 'oc new-app helloworld/jboss-eap70-deploy.json' }
+
+----
+
+.Example: Compile, build an image, and deploy to OpenShift using maven on Tekton
```
Comment (jc-berger) [inline modules/jt-examples-of-common-use-cases.adoc:55]: Similar suggestion as I made above. Consider rephrasing to: `Example: Compile and build an image and deploy it to OpenShift using maven in Tekton.`

File: modules/jt-examples-of-common-use-cases.adoc [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+            oc new-project helloworldproject
+            oc project helloworldproject
+            oc process -f jboss-eap70-binary-build.json | oc create -f -
+            oc start-build eap-helloworld-app --from-dir=artifacts/
+            oc new-app jboss-eap70-deploy.json
+
+----
+
+== Extending the core capabilities of Jenkins and Tekton by using plugins
+Jenkins has the advantage of a very rich ecosystem of numerous plugins developed over the years by its extensive user base. You can search and browse the plugins in the link:https://plugins.jenkins.io/[Jenkins Plugin Index].
```
Comment (jc-berger) [inline modules/jt-examples-of-common-use-cases.adoc:157]: `very rich` is kind of a colloquialism. How about, `Jenkins has the advantage of a large ecosystem of numerous plugins developed over the years by its extensive user base.`

**Review by jc-berger — APPROVED** [review]
/lgtm
